common/src/test/java/uk/gov/justice/services/common/converter/JsonObjectToObjectConverterTest.java

```diff
@@ -0,0 +1,137 @@
+package uk.gov.justice.services.common.converter;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.core.IsCollectionContaining.hasItems;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.doThrow;
+
+@RunWith(MockitoJUnitRunner.class)
+public class JsonObjectToObjectConverterTest {
+
+    private static final UUID ID = UUID.randomUUID();
+    private static final UUID INTERNAL_ID = UUID.randomUUID();
+    private static final String NAME = "Pojo";
+    private static final String ATTRIBUTE_1 = "Attribute 1";
+    private static final String ATTRIBUTE_2 = "Attribute 2";
+    private static final String INTERNAL_NAME = "internalName";
+    private static final String INVALID_JSON = "INVALID_JSON";
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    @Mock
+    private JsonObjectToStringConverter jsonObjectToStringConverter;
+
+    @Mock
+    private ObjectMapper mapper;
+
+    @Test
+    public void shouldConvertPojoToJsonObject() throws Exception {
+        JsonObjectToObjectConverter jsonObjectToObjectConverter = new JsonObjectToObjectConverter();
+        jsonObjectToObjectConverter.mapper = new JacksonMapperProducer().objectMapper();
+
+        JsonObject jsonObject = jsonObject();
+        Pojo pojo = jsonObjectToObjectConverter.convert(jsonObject, Pojo.class);
+
+        assertThat(pojo, notNullValue());
+        assertThat(pojo.getId(), equalTo(ID));
+        assertThat(pojo.getName(), equalTo(NAME));
+        assertThat(pojo.getAttributes(), hasItems(ATTRIBUTE_1, ATTRIBUTE_2));
+        assertThat(pojo.getInternalPojo().getInternalId(), equalTo(INTERNAL_ID));
+        assertThat(pojo.getInternalPojo().getInternalName(), equalTo(INTERNAL_NAME));
+    }
+
+    @Test
+    public void shouldThrowExceptionOnConversionError() throws IOException {
+        JsonObjectToObjectConverter jsonObjectToObjectConverter = new JsonObjectToObjectConverter();
+        jsonObjectToObjectConverter.mapper = mapper;
+        JsonObject jsonObject = jsonObject();
+
+        doThrow(IOException.class).when(mapper).writeValueAsString(jsonObject);
+        exception.expect(IllegalArgumentException.class);
```

Comment:
If we are not testing the string of the error message (or cause etc.) then I think expected exception is clearer on the Test annotation.